PhysicsTools/PatAlgos/plugins/PATElectronSlimmer.cc

```diff
@@ -72,13 +72,9 @@ pat::PATElectronSlimmer::PATElectronSlimmer(const edm::ParameterSet & iConfig) :
     reducedEndcapRecHitCollectionToken_(consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("reducedEndcapRecHitCollection"))),
     modifyElectron_(iConfig.getParameter<bool>("modifyElectrons"))
 {
-    edm::ConsumesCollector sumes(consumesCollector());
     if( modifyElectron_ ) {
       const edm::ParameterSet& mod_config = iConfig.getParameter<edm::ParameterSet>("modifierConfig");
-      electronModifier_.reset(new pat::ObjectModifier<pat::Electron>(mod_config) );
-      electronModifier_->setConsumes(sumes);
-    } else {
```

Comment:
@makortel for my understanding, this protection (here and in other PatAlgos classes) is now removed, do you consider it as redundant? The update of the code does not seem in principle to change the case that it was addressing, am I correct?

Reply:
@fabiocos 
`electronModifier_` is a `std::unique_ptr` which is by default initialized to `nullptr`, so resetting it to `nullptr` is redundant (and actually makes the code a tiny bit more difficult to reason about since one has may wonder why it is reset to `nullptr`).

Reply:
@perrotta @makortel thanks, so the protection was actually redundant even before the code update